src/index.js

```diff
@@ -1,266 +1,272 @@
+          // transform every byte to hexadecimal
+          bytes.push(byte.toString(16));
+        });
+
+        const hex = bytes.join('').toUpperCase();
+        resolve(signatures[hex]);
+      };
+
+      reader.onerror = reject;
+
+      // read first 4 bytes of sliced file as an array buffer
+      reader.readAsArrayBuffer(first4BytesOfFile);
+    });
+  }
+
+  getContentTypeFromResponseHeader () {
+    return this.request.getResponseHeader('content-type');
+  }
+
+  getContentType (file) {
+    return new Promise(async (resolve) => {
+      const { contentTypeDetermination } = this.params;
+      const defaultContentType = 'application/octet-stream';
+      let headerContentType;
+      let signatureContentType;
+
+      if (contentTypeDetermination === 'header' || contentTypeDetermination === 'full') {
+        headerContentType = this.getContentTypeFromResponseHeader();
+      }
+
+      if (contentTypeDetermination === 'signature' || contentTypeDetermination === 'full') {
+        signatureContentType = await this.getContentTypeFromFileSignature(file);
+      }
+
+      if (contentTypeDetermination === 'header') {
+        resolve(headerContentType ?? defaultContentType);
+      } else if (contentTypeDetermination === 'signature') {
+        resolve(signatureContentType ?? defaultContentType);
+      } else if (contentTypeDetermination === 'full') {
+        resolve(signatureContentType ?? headerContentType ?? defaultContentType);
+      } else {
+        resolve(defaultContentType);
+      }
+    });
+  }
+
+  createLink () {
+    let link = document.createElement('a');
+
+    link.style.display = 'none';
+    return link;
+  }
+
+  clickLink () {
+    let event;
+
+    try {
+      event = new MouseEvent('click');
+    } catch (e) {
+      event = document.createEvent('MouseEvent');
+      event.initMouseEvent('click', true, true, window, 0, 0, 0, 0, 0, false, false, false, false, 0, null);
+    }
+
+    this.link.dispatchEvent(event);
+  }
+
+  async getFile (response, fileName) {
+    const type = await this.getContentType(new Blob([response]));
+    let file;
+    let options = { type: type };
+
+    try {
+      file = new File([response], fileName, options);
+    } catch (e) {
+      file = new Blob([response], options);
+      file.name = fileName;
+      file.lastModifiedDate = new Date();
+    }
+
+    return file;
+  }
+
+  async startDownload () {
+    const fileName = this.getFileName();
+
+    const file = await this.getFile(this.request.response, fileName);
+
+    // native IE
+    if ('msSaveOrOpenBlob' in window.navigator) {
+      return window.navigator.msSaveOrOpenBlob(file, fileName);
+    }
+
+    let objectUrl = window.URL.createObjectURL(file);
+
+    this.link.href = objectUrl;
+    this.link.download = fileName;
+    this.clickLink();
+
+    setTimeout(() => {
+      (window.URL || window.webkitURL || window).revokeObjectURL(objectUrl);
+    }, 1000 * 40);
+
+    return file;

```

Comment:
Before I've made any changes this function returned `this`. Not sure if that's a mistake or on purpose, since I can't see that the returned value is used.

Reply:
yes, but the return is never used so I preferred to change in favor of a standard return instead of something mixed